pkg/controller/batch/batch_controller.go

```diff
@@ -187,7 +187,12 @@ func (r *batchResource) initializeResources() (reconcile.Result, error) {
 	}
 
 	// We update the phase separately to the spec as the status update is ignored when in the update mutate function
-	return reconcile.Result{}, r.UpdatePhase(v2.BatchInitialized, nil)
+	_, err := r.update(func() error {
```

Comment:
.SetControllerReference for Batch CR still present, see L177-L179

Reply:
Whoops, that was a result of ctrl+z when chaning my initial approach :see_no_evil: